### PR TITLE
move cocos/core/gfx to cocos/gfx

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/12647

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by minggo** [comment]
@cocos-robot run test cases

File: cocos/gfx/webgl/webgl-command-allocator.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@
  THE SOFTWARE.
  */
 
-import { CachedArray } from '../../memop/cached-array';
+import { CachedArray } from '../../core';
```
Comment (star-e) [inline cocos/gfx/webgl/webgl-command-allocator.ts:26]: Will use sub-folder reduce circular dependency?

**Comment by star-e** [comment]
Some modules in core will import from gfx and gfx will also import from core.
Why these modules not forming circular dependency?

File: cocos/gfx/webgl/webgl-command-allocator.ts (hunk 1 above)
Comment (minggo) [reply cocos/gfx/webgl/webgl-command-allocator.ts:26]: I didn't find circular dependency when compiling the engine.

**Comment by minggo** [comment]
> Some modules in core will import from gfx and gfx will also import from core. Why these modules not forming circular dependency?

I am not sure, but in final, should just import from module.

**Comment by minggo** [comment]
tested on mac on editor preview, can work

**Comment by minggo** [comment]
@cocos-robot run test cases

**Comment by shrinktofit** [comment]
Suggest title change: "move core/gfx to cocos/gfx"

**Review by PPpro — APPROVED** [review]

**Comment by minggo** [comment]
> Suggest title change: "move core/gfx to cocos/gfx"

Sounds good. Changed.

**Review by star-e — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Pull request merged**